mbed_build/_internal/mbed_tools/export.py

```diff
@@ -45,3 +47,4 @@ def export(output_directory: str, toolchain: str, mbed_target: str, project_path
     """
     cmake_file_contents = generate_cmakelists_file(mbed_target, project_path, toolchain)
     write_cmakelists_file(pathlib.Path(output_directory), cmake_file_contents)
+    click.echo(f"The program-level CMake file has been successfully exported to directory '{str(output_directory)}'")
```

Comment:
Should there be an assertion for this?

Reply:
Yes, yes there should.

Reply:
Ok, now I have written a check and pushed it, I don't like it.  I don't think I should be making checks like this in the code and should rely on python's exception mechanism. If something has gone wrong in the cmake file writing it should error and never reach the message, otherwise I think its safe to assume that its ok? Would be interested to know your take on it though.

Reply:
I'd prefer we just relied on any exceptions bubbling up if something goes wrong. I don't see why we should be peppering this top level code with `if` statements if we have a solid error handling strategy further down the stack. I'd at least move the check down into the `write_cmakelists...` function and raise an exception if the end result isn't valid.

Reply:
What I meant is assertion for `echo`. That's useful information to have and it should be guarded against removal. Sorry, should've been more explicit. ;)

Reply:
041bd43

Reply:
No need to mock, `result.output` should contain whatever is output. I think `self.assertIn` could fit here nicely if you can't do direct equality due to newlines or what not.

Reply:
My internet does not know where to send data, sorry for double comments.

Reply:
👍 b8854b5